src/state/validation.py

```diff
@@ -96,6 +96,13 @@ def wrapper(instance: C, *args: typing.Any) -> None:
                     str(exc),
                 )
                 return None
+            except ValueError as exc:
+                if defer:
+                    event, *_ = args
+                    event.defer()
+                instance.unit.status = ops.BlockedStatus(str(exc))
+                logger.exception("ValueError occurred: %s", str(exc))
+                return None
```

Comment:
a base exception like `ValueError` should not flow all the way here and should be wrapped by another Exception